actions/appSettings.ts

```diff
@@ -1,58 +1,81 @@
 'use server';
 
 import { redirect } from 'next/navigation';
+import { type z } from 'zod';
 import { safeRevalidateTag } from '~/lib/cache';
+import { appSettingSchema } from '~/schemas/appSettings';
 import { createEnvironmentFormSchema } from '~/schemas/environment';
 import { requireApiAuth } from '~/utils/auth';
 import { prisma } from '~/utils/db';
 
-export async function setAnonymousRecruitment(input: boolean) {
-  await requireApiAuth();
+// Generic function to set an app setting with validation
+export async function setAppSetting<
+  Key extends keyof z.infer<typeof appSettingSchema>,
+>(key: Key, value: z.infer<typeof appSettingSchema>[Key]) {
+  console.log('setting app setting', key, value);
+
+  // validate
+  appSettingSchema.shape[key].parse(value);
 
   await prisma.appSettings.update({
-    where: { key: 'allowAnonymousRecruitment' },
-    data: { value: input.toString() },
+    where: { key },
+    data: {
+      key: key,
+      value:
+        typeof value === 'boolean' ? value.toString() : JSON.stringify(value),
+    },
   });
 
-  safeRevalidateTag('allowAnonymousRecruitment');
+  // handle revalidation tag
+  // safeRevalidateTag('appSettings', key);
+  return value;
+}
+
+export async function createAppSetting<
```

Comment:
I'm not sure we need createAppSetting. Why can't setting just create if a record doesn't exist?

Reply:
removed and consolidated with setAppSetting using upsert [b9da84f](https://github.com/complexdatacollective/Fresco/pull/161/commits/b9da84f06fc178d6c9906a2aabf7e3958146c5d7)